### DISCUSSION TITLE
is  useSubscription is best way to get updated query in another component's useQuery with same Key?

### DISCUSSION THREAD

**Discussion by dante01yoon** [TanStack/query#4897] [Q&A]
Greetings, 
I have two Component  A, B and they share same queryClient.
AComponent call useQuery with enabled: false , and BComponent call useQuery with same key, which enable option is set true.

if AComponent need to  get update states via useQuery with same key in BComponent, not using with refetch, 
is useSubscription a best way and only way to achieve this?
```tsx
// Create a QueryClient instance
const queryClient = new QueryClient();

// Pass the queryClient instance to both components as a context provider
function App() {
  return (
    <QueryClientProvider client={queryClient}>
      <AComponent>
       <ErrorBoundary fallback={FallbackRender}>
          <BComponent />
       </ErrorBoundary>
      </AComponent>
     
    </QueryClientProvider>
  );
}

```

**Comment by dante01yoon** [comment]
My apologies I guess I misunderstood. Seems there might be no useSubscription hook. 
 then should I build another custom hook to achieve this? 🤔 

```tsx
const [data, setData] = useState();

const queryInfo = useQuery({
  queryKey: [QUERY_KEYS.EXAMPLE, params],
  queryFn: () => getExample(params),
  keepPreviousData: true,
  refetchOnWindowFocus: false,
  enabled: false,
});

useEffect(() => {
  if (queryInfo.isSuccess) {
    setData(queryInfo.data);
  }
}, [queryInfo.data]);
```

**Comment by louis-young** [comment]
Hi 👋 

Please could you expand on/clarify your question and provide a minimal runnable example that demonstrates what you're trying to achieve? I'm not entirely sure what you're asking/trying to do here.

**Comment by seiyab** [comment]
I think `useQuery` with `enabled: false` works as you expect.
Or do I misunderstand?
Here is a reproduction.
https://codesandbox.io/s/cocky-tom-ndkxzi
